src/primitives/block.cpp

```diff
@@ -11,55 +11,58 @@
 #include <crypto/common.h>
 
 // yespower
-#include <stdlib.h>
 #include <crypto/yespower-1.0.1/yespower.h>
 #include <streams.h>
 #include <version.h>
 
-// yespower cache
+// yespower exit()
+#include <stdlib.h>
+
+// yespower PoW cache
 #include <sync.h>
 
 uint256 CBlockHeaderUncached::GetHash() const
 {
     return SerializeHash(*this);
 }
 
-// yespower
+// yespowerUncached
 uint256 CBlockHeaderUncached::GetPoWHash() const
 {
-    uint256 thash;
-    CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
-    ss << *this;
-    yespower_params_t yespower_1_0_sugarchain = {
+    static const yespower_params_t yespower_1_0_sugarchain = {
         .version = YESPOWER_1_0,
         .N = 2048,
         .r = 32,
         .pers = (const uint8_t *)"Satoshi Nakamoto 31/Oct/2008 Proof-of-work is essentially one-CPU-one-vote",
         .perslen = 74
     };
-    if (yespower_tls( (unsigned char *)&ss[0], ss.size(), &yespower_1_0_sugarchain, (yespower_binary_t *)&thash) ) {
-        abort();
+    uint256 hash;
+    CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
+    ss << *this;
+    if (yespower_tls((const uint8_t *)&ss[0], ss.size(), &yespower_1_0_sugarchain, (yespower_binary_t *)&hash)) {
+        fprintf(stderr, "Error: CBlockHeaderUncached: failed to compute PoW hash (out of memory?)\n");
+        exit(1);
     }
-    return thash;
+    return hash;
 }
 
-
+// yespower_cached
 uint256 CBlockHeader::GetPoWHash_cached() const
 {
     uint256 block_hash = GetHash();
     LOCK(cache_lock);
     if (cache_init) {
         if (block_hash != cache_block_hash) {
-            fprintf(stderr, "Error: CBlockHeader: block hash changed unexpectedly\n");
+            fprintf(stderr, "Error: GetPoWHash_cached: block hash changed unexpectedly\n");
```

Comment:
should `CBlockHeader_cached`

Reply:
https://github.com/sugarchain-project/sugarchain/pull/69